src/Schema/Directives/ExtendsDirective.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace Nuwave\Lighthouse\Schema\Directives;
+
+use GraphQL\Language\AST\NodeList;
+use GraphQL\Language\AST\TypeDefinitionNode;
+use Nuwave\Lighthouse\Exceptions\DefinitionException;
+use Nuwave\Lighthouse\Schema\AST\DocumentAST;
+use Nuwave\Lighthouse\Support\Contracts\TypeManipulator;
+
+class ExtendsDirective extends BaseDirective implements TypeManipulator
+{
+    public static function definition(): string
+    {
+        return /* @lang GraphQL */ <<<'GRAPHQL'
+"""
+This allows types to extends other Types
+
+it inherits properties from the parent type. 
+It overides same attributes present in the parent type.
+"""
+directive @extends(
+    """
+    The parent Type
+    """
+    parent: String!
+) on OBJECT
+GRAPHQL;
+    }
+
+    /**
+     * @param   \Nuwave\Lighthouse\Schema\AST\DocumentAST         $documentAST
+     * @param   \GraphQL\Language\AST\TypeDefinitionNode  $typeDefinition
+     *
+     * @return  void
+     */
+    public function manipulateTypeDefinition(DocumentAST &$documentAST, TypeDefinitionNode &$typeDefinition)
+    {
+        $parent_name = $this->getMethodArgumentParts('parent')[0];
+
+        $parent = $documentAST->types[$parent_name];
```

Comment:
This should check and throw a developer friendly exception if the type does not exist.

Reply:
Suggestions?

Reply:
I am not sure what else I can suggest, what part of my comment was not clear?